ide-support/revsaveasiosstandalone.livecodescript

```diff
@@ -747,21 +747,98 @@ end revSaveAsMobileDeployer
+         "retina icon,iphone,57x57,2x,Icon-App-57x57@2x.png" & return & \
+         "iOS 7 retina icon,iphone,60x60,2x,Icon-App-60x60@2x.png" & return & \
+         "iphone 6 plus icon,iphone,60x60,3x,Icon-App-60x60@3x.png" & return & \
+         "iphone X icon,iphone,60x60,3x,Icon-App-60x60@3x.png" & return & \
+         "ipad icon,ipad,72x72,1x,Icon-App-72x72@1x.png" & return & \
+         "ipad retina icon,ipad,72x72,2x,Icon-App-72x72@2x.png" & return & \
+         "iOS 7 ipad icon,ipad,76x76,1x,Icon-App-76x76@1x.png" & return & \
+         "iOS 7 ipad retina icon,ipad,76x76,2x,Icon-App-76x76@2x.png" & return & \
+         "iPad Pro 12.9 icon,ipad,83.5x83.5,2x,Icon-App-83.5x83.5@2x.png" \
+         into sIconsList
+   
+   local tAssetsFolder, tAppIconFolder
+   put tempname() into tAssetsFolder
+   create folder  tAssetsFolder
+   put there is a folder tAssetsFolder
+   put tAssetsFolder & slash & "AppIcon.appiconset" into tAppIconFolder
+   create folder tAppIconFolder
+   put there is a folder tAppIconFolder
+   
+   local tContentsJSON
+   put "{" & CR & quote & "images" & quote & ":[" & CR after tContentsJSON
+   
+   repeat for each line tType in sIconsList
       get pSettings["ios," & item 1 of tType]
       if not (it begins with "/") then
          put pBaseFolder & slash before it
       end if
       if there is a file it then         
-         put url ("binfile:" & it) into url ("binfile:" & pAppBundle & slash & "Icon" & item 2 of tType & ".png")
+         put url ("binfile:" & it) into url ("binfile:" & tAppIconFolder & slash &  item 5 of tType)
          if the result is not empty then
             throw "could not copy icon '" & it & "'"
          end if
+         -- add an entry in the Contents.json file
+         local tEntry
+         put "{" & CR after tEntry
+         put quote & "idiom" & quote & ":" & quote & item 2 of tType & quote & comma & CR after tEntry
+         put quote & "size" & quote & ":" & quote & item 3 of tType & quote & comma & CR after tEntry
+         put quote & "scale" & quote & ":" & quote & item 4 of tType & quote & comma & CR after tEntry
+         put quote & "filename" & quote & ":" & quote & item 5 of tType & quote  & CR after tEntry
+         put "}" & comma & CR after tEntry
+         put tEntry after tContentsJSON
       end if
    end repeat
+   
+   put "]" & CR & "}" after tContentsJSON
+   put tContentsJSON into url ("file:" & tAppIconFolder & slash &  "Contents.json")
+   if the result is not empty then
+      throw "could not create json file '" & it & "'"
+   end if
+   
+   local tCompiledAssetsFolder
+   put tempname() into tCompiledAssetsFolder
+   create folder  tCompiledAssetsFolder
+   
+   -- create the Assets.car file
+   get shell("xcrun actool " &  tAssetsFolder & " --compile " & tCompiledAssetsFolder & " --platform iphoneos --minimum-deployment-target 9.0 --app-icon AppIcon --output-partial-info-plist " & tCompiledAssetsFolder &"/partial.plist")
```

Comment:
could you put a couple of line continuations here to make this a bit more readable?

Reply:
sure :)